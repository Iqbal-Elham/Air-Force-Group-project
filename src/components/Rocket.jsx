/* eslint-disable */
import React from "react";
import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import 'bootstrap/dist/css/bootstrap.min.css';
import { Button, Badge } from 'react-bootstrap';
import { setReserved, cancelReserved } from "../redux/rockets/rocketsSlice";

const Rocket = ({ id, name, type, img, description, reserved }) => {
  console.log(reserved)
  const dispatch = useDispatch();
  return (
    <div className="d-flex p-3 gap-3">
      <img src={img} alt="Rocket" width="470" height="350" className="img-fluid" />
      <div className="pt-1">
        <h2>{name}</h2>
        <p className="text-justify h5"><span>{reserved ? <Badge>Reserved</Badge> : ''} </span>{description}</p>
        <p className="h6">{type}</p>
        <div>
        {
          !reserved ? <Button 
          className="mt-4 py-2 px-4" 
          variant="primary" 
          onClick={() => dispatch(setReserved(id))}>
            Reserve Rocket
          </Button>
          : <Button 
          className="mt-4 py-2 px-4" 
          variant="secondary" 
          onClick={() => dispatch(cancelReserved(id))}>
            Cancel Reservation
          </Button>
        }
        </div>
      </div>
    </div>
  );
};

Rocket.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  img: PropTypes.array.isRequired,
};

export default Rocket;
