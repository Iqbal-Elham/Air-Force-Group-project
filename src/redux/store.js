import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
});

export default store;
