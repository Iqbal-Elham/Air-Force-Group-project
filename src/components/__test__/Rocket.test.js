import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rocket from '../Rocket';
import store from '../../redux/store';

describe('Rockets component', () => {
  it('should match to snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
