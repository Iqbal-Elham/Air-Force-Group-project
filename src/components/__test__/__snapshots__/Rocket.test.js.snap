// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component should match to snapshot 1`] = `
<div
  className="d-flex p-3 gap-3"
>
  <img
    alt="Rocket"
    className="img-fluid"
    height="350"
    width="470"
  />
  <div
    className="pt-1"
  >
    <h2 />
    <p
      className="text-justify h5"
    >
      <span>
         
      </span>
    </p>
    <p
      className="h6"
    />
    <div>
      <button
        className="mt-4 py-2 px-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
